### DISCUSSION TITLE
Game runs out of memory, crashes on Mac M1

### DISCUSSION THREAD

**Discussion by andrewpey** [voxeltycoon/issues#924] [Troubleshooting] [answered]
### Original issue
- https://github.com/voxeltycoon/issues/issues/918

### Description

There're several reports that the game eats a lot of memory and then crashes on Mac M1 after the 0.87 update.

While we keep investigating the issue, here're temporary solutions to this problem:

### Temporary solution 1

Try to change your game resolution to something other than your native screen resolution (or _Looks like_ resolution if you use scaling).

### Temporary solution 2

Switch to the `m1_test` branch on Steam. It should work on M1 without issues. The only downside is that build on that branch is locked to older version of the game.

To switch to the branch, do the following: 

- Right-click Voxel Tycoon on Steam
- Properties
- Betas
- Select `m1_test` branch in the betas list

**Comment by andrewpey** [comment]
We've released an `0.87.1.5` update that should fix this issue.

**Reply by andrewpey** [reply]
Seems like the issue is still here: https://steamcommunity.com/app/732050/eventcomments/3176733459032297101

**Comment by rbranham** [comment]
Ran into same issue on my M1 mac. Just to add some additional information, it looks like in addition to the memory spiking the ports(internal mac ones, not network ones) ramp up infinitely as well. Not sure which is cause and which is the effect but might be useful to know. Also activity monitor says I'm using 0 swap so I'm not sure where this memory is leaking to?

**Comment by RoeyPrat** [comment]
Hi, I got the same issue on a mac that is not M1.
Memory eaten, ports taken.
```
  Model Identifier:	MacBookPro16,2
  Processor Name:	Quad-Core Intel Core i7
```
I'm on version 0.87.1.5.

**Comment by andrewpey** [comment]
We've just pushed a new build to the [beta branch](https://voxeltycoon.xyz/devlog/beta-branch/) where we upgraded to the most recent stable Unity version (2021.3.0). Please try it out and let us know if the issue is still here.

**Comment by andrewpey** [comment]
We've just pushed a new build (`0.87.1.8`) to the [beta branch](https://voxeltycoon.xyz/devlog/beta-branch/) which includes a native M1 version. Please try it out and let us know if the issue is still here (and if it works at all).

**Reply by rbranham** [reply]
Was able to test for a couple of hours on the new branch, memory problem seems to be fixed. Thanks!

**Comment by andrewpey** [comment]
We've added a second temporary solution involving just changing your in-game resolution (see the original post).

**Comment by andrewpey** [comment, marked as answer]
The recent update (0.88.3.1) contains a possible fix for this issue. Please give it a try and let us know if it helps.
